react/features/jwt/middleware.js

```diff
@@ -96,7 +100,8 @@ function _maybeSetCallOverlayVisible({ dispatch, getState }, next, action) {
             default: {
                 // The CallOverlay it to no longer be displayed/visible as soon
                 // as another participant joins.
-                callOverlayVisible = isLonelyCall(state);
+                callOverlayVisible = getParticipantCount(state) === 1
+                    && Boolean(getLocalParticipant(state));
```

Comment:
In all honestly, I don't know either. I just didn't want to change existing behavior in the file. In any case, I think if we go with the route of using some kind of filmstrip state and not reusing participants for filmstrip, the fake user can be avoided and these changes to middleware won't be needed.